pep.rst

```diff
@@ -1,13 +1,41 @@
+PEP: 9999
+Title: Tag Strings For Writing Domain-Specific Languages
+Author: Jim Baker <jim.baker@python.org>, Guido van Rossum <guido@python.org>, Paul Everitt <pauleveritt@me.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 01-Jun-2024
+Python-Version: 3.13
+
 Abstract
 ========
 
 This PEP introduces tag strings for custom, repeatable string processing. Tag strings
 are an extension to f-strings, with a custom function -- the "tag" -- in place of the
-`f` prefix. This function can then provide rich features such as safety checks, lazy
-evaluation, domain specific languages (DSLs) for web templating, and more.
+``f`` prefix. This function can then provide rich features such as safety checks, lazy
+evaluation, domain-specific languages (DSLs) for web templating, and more.
+
+Tag strings are similar to `JavaScript tagged template literals <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates>`_
+and related ideas in other languages. The following tag string usage shows how similar it is to an ``f`` string, albeit
+with the ability to process the literal string and embedded values:
+
+.. code-block:: python
+
+    name = "World"
+    greeting = greet"hello {name}"
+    assert greeting == "Hello WORLD!"
+
 
-Tag strings are similar to `JavaScript tagged templates <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates>`_
-and similar ideas in other languages.
+Tag functions accept prepared arguments and return a string:
+
+.. code-block:: python
+    def greet(*args):
+        salutation, recipient, *_ = args
+        _, getvalue = recipient
+        return f"{salutation.title()} {getvalue().upper()}!"
```

Comment:
I think I missed an extra `strip()` here. That, or removing the space. I might argue against the `strip()` just to have one less thing for the user to think about, but either way:

```suggestion
        return f"{salutation.title()}{getvalue().upper()}!"
```
Or
```suggestion
        return f"{salutation.title().strip()} {getvalue().upper()}!"
```